libs/src/utils/fileHasher.ts

```diff
@@ -0,0 +1,278 @@
+
+// Base functionality for only hash logic taken from https://github.com/alanshaw/ipfs-only-hash/blob/master/index.js
+
+export type Content = NodeJS.ReadableStream | Buffer | string
+export interface ImageHasher {
+  options: UserImporterOptions
+  content: ImportCandidate
+}
+export interface NonImageHasher {
+  options: UserImporterOptions
+  content: Uint8Array
+}
+export interface HashedImage {
+  path: string | undefined
+  cid: string
+  size: number
+}
+
+const block: Blockstore = {
+  get: async (key: CID, _options?: Options) => {
+    throw new Error(`unexpected block API get for ${key}`)
+  },
+  put: async (_key: CID, _val: Uint8Array, _options?: Options) => {
+    throw new Error('unexpected block API put')
+  },
+  open: async function (): Promise<void> {
+    throw new Error('Function not implemented.')
+  },
+  close: async function (): Promise<void> {
+    throw new Error('Function not implemented.')
+  },
+  has: async function (_key: CID, _options?: Options): Promise<boolean> {
+    throw new Error('Function not implemented.')
+  },
+  delete: async function (_key: CID, _options?: Options): Promise<void> {
+    throw new Error('Function not implemented.')
+  },
+  putMany: function (
+    _source: AwaitIterable<Pair<CID, Uint8Array>>,
+    _options?: Options
+  ): AsyncIterable<Pair<CID, Uint8Array>> {
+    throw new Error('Function not implemented.')
+  },
+  getMany: function (
+    _source: AwaitIterable<CID>,
+    _options?: Options
+  ): AsyncIterable<Uint8Array> {
+    throw new Error('Function not implemented.')
+  },
+  deleteMany: function (
+    _source: AwaitIterable<CID>,
+    _options?: Options
+  ): AsyncIterable<CID> {
+    throw new Error('Function not implemented.')
+  },
+  batch: function (): Batch<CID, Uint8Array> {
+    throw new Error('Function not implemented.')
+  },
+  query: function (
+    _query: Query<CID, Uint8Array>,
+    _options?: Options
+  ): AsyncIterable<Pair<CID, Uint8Array>> {
+    throw new Error('Function not implemented.')
+  },
+  queryKeys: function (
+    _query: KeyQuery<CID>,
+    _options?: Options
+  ): AsyncIterable<CID> {
+    throw new Error('Function not implemented.')
+  }
+}
+
+export const fileHasher = {
+  convertNanosToMillis(nanoSeconds: bigint) {
+    return nanoSeconds / BigInt(1000000)
+  },
+
+  /**
+   * Used to iniitalize the only hash fns. See Alan Shaw's reference code for more context.
+   */
+  initImageHasher(
+    content: ImportCandidate,
+    options: UserImporterOptions
+  ): ImageHasher {
+    options = options || {}
+    options.onlyHash = true
+    options.cidVersion = 0
+
+    return { options, content }
+  },
+
+  /**
+   * Used to iniitalize the only hash fns. See Alan Shaw's reference code for more context.
+   */
+  initNonImageHasher(
+    content: Uint8Array,
+    options: UserImporterOptions
+  ): NonImageHasher {
+    options = options || {}
+    options.onlyHash = true
+    options.cidVersion = 0
+
+    return { options, content }
```

Comment:
do we not need https://github.com/AudiusProject/audius-protocol/blob/e1f9ac261aac6a1bb6ba1b850f8c0ce71888bc56/creator-node/src/fileHasher.js#L153 anymore?

Reply:
we don't and I think the reason we had it is that it was in an example from a different repo. in that example they didn't convert to a buffer first. since we're always converting to a buffer for non-images, the result can never be a string (the logic before this gets called checks for a string already and reads from a file if it's a string)

Reply:
and since we have a test for that should be okay 🙂